### DISCUSSION TITLE
Elevation data for final project error 400

### DISCUSSION THREAD

**Discussion by taylor-obrien** [earthlab-education/Earth-Analytics-2023-01-Intro#654] [Discussion submissions]
I am trying to download the elevation data from Appears for the final project, my code looks like this.

![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/143227722/a01f4758-c674-435f-a0e9-81116a904161)

However, I keep getting this error, what am I doing wrong here?

![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/143227722/3835a250-dbcd-4ab1-95a3-877ff99e5a1c)

**Comment by NobodyLivesHere** [comment]
There is something it doesn't like in how it's submitting the request to the MODIS server that's making it throw a 400 error.  It might be that it isn't seeing the geometry right for the study site or something funky for the date.

**Reply by taylor-obrien** [reply]
Thanks! I'm hoping someone knows where I'm going wrong or how to correct this issue.

**Comment by pth6570** [comment]
>
I’m not sure if either will fix the issue but a couple of things you might want to try:

1 - I have my layer name as “SRTMGL1_DEM” instead of “SRTMGL1_DEM Elevation”.

2 - If you look at the AppEEARS API Documentation (https://appeears.earthdatacloud.nasa.gov/api/#task-object), it looks like it might want specific date formats based on whether you’re using a recurring or non-recurring date range. Since you have recurring set to False, perhaps try changing the start and end dates to MM-DD-YYYY, and dropping year_range?

**Comment by edchanonline** [comment]
@taylor-obrien 
I got it working with layer `SRTMGL1_NUM` instead of `SRTMGL1_DEM`.

You can list the layer with this snippet of Python code:
```
import requests
import json

product_id = 'SRTMGL1_NUMNC.003'
response = requests.get('https://appeears.earthdatacloud.nasa.gov/api/product/{0}'.format(product_id))
layer_response = response.json()
print(json.dumps(layer_response))
```

Response (Layer Info):
```
{
    "SRTMGL1_NUM":
    {
        "AddOffset": "",
        "Available": true,
        "DataType": "int16",
        "Description": "Quality Assurance",
        "Dimensions":
        [
            "time",
            "lat",
            "lon"
        ],
        "FillValue": 0,
        "FillValueRaw": "0",
        "Group": "",
        "IsQA": true,
        "Layer": "SRTMGL1_NUM",
        "OrigDataType": "int16",
        "OrigValidMax": 255,
        "OrigValidMin": 0,
        "QualityLayers": "",
        "QualityProductAndVersion": "",
        "ScaleFactor": "",
        "Units": "",
        "ValidMax": 255,
        "ValidMin": 0,
        "XSize": 3601,
        "YSize": 3601
    }
}
```

My sample download code:
```
elev_downloader = etapp.AppeearsDownloader(
    ea_dir=srtm_dir,
    product='SRTMGL1_NC.003',
    layer='SRTMGL1_DEM',
    start_date="02-11-2000",
    end_date="02-21-2000",
    polygon=filtered_gpd
)

elev_downloader.download_files()
```

**Reply by eculler** [reply]
Hey @eggvoice @taylor-obrien the NUM layer is the quality assurance, not the DEM! But, the second sample download code here does look right. I ran into the same problem last week, and it was the layer name for me too.

**Reply by edchanonline** [reply]
Thank you! I just figure that out too!! 
I was mixing up my product ids and layer ids as I was trying to figure out which combos would work.

```
product_id = 'SRTMGL1_NC.003'
layer_id = 'SRTMGL1_DEM'

srtm_downloader = etapp.AppeearsDownloader(
    ea_dir=data_dir,
    download_key=download_key, # srtm-pawnee
    product=product_id,
    layer=layer_id,
    start_date='02-11-2000',
    end_date='02-21-2000',
    polygon=pawnee_gdf
)
if not os.path.exists(os.path.join(data_dir, download_key)):
    srtm_downloader.download_files(cache=True)
```

**Comment by nbol48** [comment]
I am running into the same error from my code. Is it possible that this download won't work in a function? I tried it outside of the function as well and couldn't get rid of the error so it seems unlikely but hopefully some more eyes will see something I'm missing! 

'''
def srtm_download(desired_file_name, gdf):
    """
    Parameters
    ==========
    file_name : str
        Desired file name for the DataArray when it is saved in the earthpy downloads directory
    gdf : GeoDatafram
        The dataframe conatining a single geometry for the APPEEARS downloader to indentify 

    Returns
    =======
    DataArray of the elevations raster data for the selected national grassland geometry
    """
    # download_key = desired_file_name
    srtm_downloader = eaap.AppeearsDownloader(
        download_key=desired_file_name,
        product='SRTMGL1_NC.003',
        layer='STRMGL1_DEM',
        start_date='02-11-2000',
        end_date='02-21-2000',
        polygon=gdf
    )
    srtm_downloader.download_files()

    srtm_filepaths = glob(os.path.join(srtm_downloader.data_dir,
                                       'SRTMGL1_NC.003*',
                                       '*.tif'
                                       ))
    return [rxr.open_rasterio(srtm_filepaths, masked=True).squeeze() for srtm_filepaths in srtm_filepaths][0]
'''

**Reply by nbol48** [reply]
This is the message I get from the link the error provides: {"message": "You don't have the permission to access the requested resource. It is either read-protected or not readable by the server."}

**Reply by nbol48** [reply]
......and I got it. STRM vs SRTM.
